ceno_zkvm/src/scheme/hal.rs

```diff
@@ -6,31 +6,13 @@ use crate::{
     structs::{TowerProofs, ZKVMProvingKey},
 };
 use ff_ext::ExtensionField;
+use gkr_iop::hal::ProverBackend;
 use mpcs::{Point, PolynomialCommitmentScheme};
 use multilinear_extensions::{mle::MultilinearExtension, util::ceil_log2};
 use sumcheck::structs::IOPProverMessage;
 use transcript::Transcript;
 use witness::next_pow2_instance_padding;
 
-pub trait MultilinearPolynomial<E: ExtensionField> {
-    fn num_vars(&self) -> usize;
-    fn eval(&self, point: Point<E>) -> E;
-}
-
-/// Defines basic types like field, pcs that are common among all devices
-/// and also defines the types that are specific to device.
-pub trait ProverBackend {
```

Comment:
`ProverBackend` was moved under gkr-iop package, as both ceno_zkvm and gkr-iop depends on it